Improve error display

tanchekwei opened the pull request:
Display the error content instead of `Error: [object CustomEvent]`

![image](https://github.com/sunner/ChatALL/assets/26683979/d9230a4c-1da0-4ee2-a646-a6891e910c3a)

tanchekwei commented:
Should I apply this changes to other bots as well?

sunner commented:
Looks good. To apply all, how about `reject(event);` here and unwrap it at https://github.com/sunner/ChatALL/blob/main/src/bots/Bot.js#L184C4-L187C4 ?

tanchekwei commented:
After I tried, I found that the `event.source.xhr` will became `null` when unwrap in Bot.js.

The reason of this is because of sse.js explicitly set it to `null` after stream is closed, in this line: https://github.com/mpetazzoni/sse.js/blob/cfe70a36d49009e403e1cd2419b097497ab693da/lib/sse.js#L209

So, I have created `getSSEDisplayError` in Bot.js and call the function to get the error content before sse.js set `xhr` to `null`. 

Only found Pi, YouChat and QianWen only used `reject(new Error(event));`

Other bots already have their own way of returning errors so I left them unchanged.

Also updated to wrap error message in [collapsed section](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections#creating-a-collapsed-section)

![image](https://github.com/sunner/ChatALL/assets/26683979/aae6db22-25da-401a-9bac-ee955e7ecb81)

sunner commented:
The error message in light mode is not clear:

<img width="965" alt="image" src="https://github.com/sunner/ChatALL/assets/255413/57a292ab-af33-45eb-bfef-b7b155452311">

tanchekwei commented:
Oops, I forgot to check the light mode.

Added `background-color: inherit;`
https://github.com/sunner/ChatALL/blob/ae2b7da343bd22fee4bfeb71242c30dda0ae398c/src/components/Messages/ChatResponse.vue#L507

![image](https://github.com/sunner/ChatALL/assets/26683979/27d4f392-59af-46b5-99bd-e08da4f320ea)

Rebased and solved conflict

sunner commented:
Thank you. This makes ChatALL much better.
